File: backend/app/schemas/task.py
"""Task schemas."""
from datetime import datetime, date
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class TaskCreate(BaseModel):
    """Schema for creating a task."""
    title: str
    description: Optional[str] = None
    priority: Optional[str] = "medium"
    assignee: Optional[str] = None
    due_date: Optional[date] = None
    tags: Optional[list[str]] = None
    parent_task_id: Optional[UUID] = None


class TaskUpdate(BaseModel):
    """Schema for updating a task."""
    title: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None
    priority: Optional[str] = None
    assignee: Optional[str] = None
    due_date: Optional[date] = None
    tags: Optional[list[str]] = None


class TaskResponse(BaseModel):
    """Schema for task response."""
    id: UUID
    project_id: UUID
    parent_task_id: Optional[UUID] = None
    title: str
    description: Optional[str] = None
    status: str
    priority: str
    assignee: Optional[str] = None
    due_date: Optional[date] = None
    tags: Optional[list[str]] = None
    created_at: datetime
    updated_at: Optional[datetime] = None
    
    model_config = ConfigDict(from_attributes=True)


class TaskListResponse(BaseModel):
    """Schema for list of tasks."""
    tasks: list[TaskResponse]
    total: int


class BulkTaskCreate(BaseModel):
    """Schema for bulk creating tasks."""
    tasks: list[TaskCreate]


class BulkTaskResponse(BaseModel):
    """Schema for bulk task creation response."""
    created: list[TaskResponse]
    count: int

File: frontend/src/types/task.ts
export type TaskStatus = 'todo' | 'in_progress' | 'review' | 'done';
export type TaskPriority = 'low' | 'medium' | 'high' | 'critical';

export interface Task {
  id: string;
  project_id: string;
  parent_task_id: string | null;
  title: string;
  description: string | null;
  status: TaskStatus;
  priority: TaskPriority;
  assignee: string | null;
  due_date: string | null;
  tags: string[] | null;
  created_at: string;
  updated_at: string | null;
}

export interface TaskCreate {
  title: string;
  description?: string;
  priority?: TaskPriority;
  assignee?: string;
  due_date?: string;
  tags?: string[];
  parent_task_id?: string;
}

export interface TaskUpdate {
  title?: string;
  description?: string;
  status?: TaskStatus;
  priority?: TaskPriority;
  assignee?: string;
  due_date?: string;
  tags?: string[];
}

export interface TaskListResponse {
  tasks: Task[];
  total: number;
}

export interface BulkTaskCreate {
  tasks: TaskCreate[];
}

export interface BulkTaskResponse {
  created: Task[];
  count: number;
}
